index.js

```diff
@@ -66,6 +66,10 @@ module.exports = function (val, opts, pad) {
 			objKeys = Object.keys(val);
 
 			var ret = '{\n' + objKeys.map(function (el, i) {
+				if (opts.filter && !opts.filter.call(this, val, el)) {
```

Comment:
Why are you using `.call`?

`val` should probably be `val[el]`. I don't see the point of returning the whole object, when the point is to check the current object.